Sources/Datadog/RUM/RUMMonitor/Scopes/RUMSessionScope.swift

```diff
@@ -0,0 +1,111 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+
+internal class RUMSessionScope: RUMScope {
+    struct Constants {
+        /// If no interaction is registered within this period, a new session is started.
+        static let sessionTimeoutDuration: TimeInterval = 15 * 60 // 15 minutes
+        /// Maximum duration of a session. If it gets exceeded, a new session is started.
+        static let sessionMaxDuration: TimeInterval = 4 * 60 * 60 // 4 hours
+    }
+
+    // MARK: - Initialization
+
+    unowned let parent: RUMScope
+    private let dependencies: RUMScopeDependencies
+
+    /// This session UUID.
+    private var sessionUUID: UUID
+    /// The start time of this session.
+    private var sessionStartTime: Date
+    /// Time of the last RUM interaction noticed by this session.
+    private var lastInteractionTime: Date
+
+    init(
+        parent: RUMScope,
+        dependencies: RUMScopeDependencies
+    ) {
+        self.parent = parent
+        self.dependencies = dependencies
+        self.sessionUUID = UUID()
+        self.sessionStartTime = dependencies.dateProvider.currentDate()
+        self.lastInteractionTime = self.sessionStartTime
+    }
+
+    // MARK: - RUMScope
+
+    var context: RUMContext {
+        var context = parent.context
+        context.sessionID = sessionUUID
+        return context
+    }
+
+    func process(command: RUMCommand) -> Bool {
+        if timedOutOrExpired() {
+            return true // end session
```

Comment:
this line should return `false` according to RFC

Reply:
👍 good catch, done ✅. The RFC says `false` in one place, but then keeps saying `true` in code commends. I will update it as well.